pkg/shell/server/shell_server.go

```diff
@@ -84,6 +84,10 @@ func (sh shell) listener(ctx context.Context, health cell.Health) error {
 	for ctx.Err() == nil {
 		conn, err := l.Accept()
 		if err != nil {
+			// If context is cancelled, the listener was closed gracefully
+			if ctx.Err() != nil {
```

Comment:
Probably better to be a bit more specific and use `if errors.Is(err, context.Canceled)`, just in case.